### ISSUE TITLE
improve optimizing + callback

### ISSUE THREAD

**Issue by fiburg** [fiburg/bdp_cnn#29]
Added callback, which reduces the learning rate when a metric has stopped improving: 
`ReduceLROnPlateau(monitor='val_loss', factor=0.2, patience=5)` 

According to this I explored the two optimizers `adam` and `Adadelta`.

With the callback `History`, we can observe the loss and validation loss for each epoch. A plotting routine was added in our `Evaluater`.
